src/node_locks.cc

```diff
@@ -0,0 +1,880 @@
+                                               Local<Context> context,
+                                               const std::u16string& name,
+                                               Lock::Mode mode,
+                                               const std::string& client_id);
+
+Lock::Lock(Environment* env,
+           const std::u16string& name,
+           Mode mode,
+           const std::string& client_id,
+           Local<Promise::Resolver> waiting,
+           Local<Promise::Resolver> released)
+    : env_(env), name_(name), mode_(mode), client_id_(client_id) {
+  waiting_promise_.Reset(env_->isolate(), waiting);
+  released_promise_.Reset(env_->isolate(), released);
+}
+
+LockRequest::LockRequest(Environment* env,
+                         Local<Promise::Resolver> waiting,
+                         Local<Promise::Resolver> released,
+                         Local<Function> callback,
+                         const std::u16string& name,
+                         Lock::Mode mode,
+                         const std::string& client_id,
+                         bool steal,
+                         bool if_available)
+    : env_(env),
+      name_(name),
+      mode_(mode),
+      client_id_(client_id),
+      steal_(steal),
+      if_available_(if_available) {
+  waiting_promise_.Reset(env_->isolate(), waiting);
+  released_promise_.Reset(env_->isolate(), released);
+  callback_.Reset(env_->isolate(), callback);
+}
+
+bool LockManager::IsGrantable(const LockRequest* request) const {
+  // Steal requests bypass all normal granting rules
+  if (request->steal()) return true;
+
+  auto held_locks_iter = held_locks_.find(request->name());
+  // No existing locks for this resource name
+  if (held_locks_iter == held_locks_.end()) return true;
+
+  // Exclusive requests cannot coexist with any existing locks
+  if (request->mode() == Lock::Mode::Exclusive) return false;
+
+  // For shared requests, check if any existing lock is exclusive
+  for (const auto& existing_lock : held_locks_iter->second) {
+    if (existing_lock->mode() == Lock::Mode::Exclusive) return false;
+  }
+  // All existing locks are shared, so this shared request can be granted
+  return true;
+}
+
+// Called when the user callback promise fulfills
+static void OnLockCallbackFulfilled(
+    const v8::FunctionCallbackInfo<v8::Value>& info) {
+  HandleScope handle_scope(info.GetIsolate());
+  Environment* env = Environment::GetCurrent(info);
+
+  // Extract the LockHolder from V8 External data
+  auto* lock_holder =
+      static_cast<LockHolder*>(info.Data().As<External>()->Value());
+  std::shared_ptr<Lock> lock = lock_holder->lock();
+  delete lock_holder;
+
+  // Release the lock and continue processing the queue.
+  LockManager::GetCurrent()->ReleaseLockAndProcessQueue(
+      env, lock, info[0], false);
+}
+
+// Called when the user callback promise rejects
+static void OnLockCallbackRejected(
+    const v8::FunctionCallbackInfo<v8::Value>& info) {
```

Comment:
Here and other places ... 

```suggestion
    const FunctionCallbackInfo<Value>& info) {
```

Reply:
All `v8::…` have been replaced with their shorter aliases